src/Optimization/enums.jl

```diff
@@ -0,0 +1 @@
+@scoped_enum(BuildStatus, IN_PROGRESS = -1, BUILT = 0, FAILED = 1, EMPTY = 2,)
```

Comment:
Should RunStatus be here also?

Reply:
I didn't because the RUN_STATUS are different in PSI and PSID. They have different definitions.

Reply:
Are you referring to this enum in PSID? There is no `RUN_STATUS` or `RunStatus`.
```
@enum BUILD_STATUS begin
    BUILT = 0
    BUILD_IN_PROGRESS = 1
    BUILD_INCOMPLETE = 2
    BUILD_FAILED = 3
    SIMULATION_INITIALIZED = 4
    SIMULATION_STARTED = 5
    SIMULATION_FINALIZED = 6
    SIMULATION_FAILED = 7
end
```
 I think this means that we can move both `BuildStatus` and `RunStatus` from PSI to IS.Optimization.

Reply:
In this case we should have a IS.Simulation submodule. I think in PSID not having two Status ENUMS is an oversight

Reply:
[8c62ffb](https://github.com/NREL-Sienna/InfrastructureSystems.jl/pull/335/commits/8c62ffbdf3162440e1524f16cef9cfe3c7d236fe)